frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/RediscoverSuggester.java

```diff
@@ -108,15 +113,35 @@ public void run(ProgressIndicator indicator) {
 
   private boolean areAllLocalBranchesManaged(Path macheteFilePath) {
     val localBranches = gitRepository.getBranches().getLocalBranches();
-    val branchLayoutReader = RuntimeBinding.instantiateSoleImplementingClass(IBranchLayoutReader.class);
     try {
       val branchLayout = branchLayoutReader.read(macheteFilePath);
       val localBranchNames = List.ofAll(localBranches)
           .map(GitReference::getName);
 
       return localBranchNames.forAll(branchLayout::hasEntry);
-    } catch (BranchLayoutException e) {
-      return false;
+    } catch (BranchLayoutException ignored) {}
+
+    return false;
+  }
+
+  private boolean isBranchLayoutTheSame(Path macheteFilePath) {
+    Path rootDirPath = gitRepository.getRootDirectoryPath().toAbsolutePath();
+    Path mainGitDirPath = gitRepository.getMainGitDirectoryPath().toAbsolutePath();
+    Path worktreeGitDirPath = gitRepository.getWorktreeGitDirectoryPath().toAbsolutePath();
+
+    val discoverRunResult = Try.of(() -> RuntimeBinding.instantiateSoleImplementingClass(IGitMacheteRepositoryCache.class)
+        .getInstance(rootDirPath, mainGitDirPath, worktreeGitDirPath).discoverLayoutAndCreateSnapshot());
+
+    if (discoverRunResult.isSuccess()) {
+      try {
+        val managedBranchLayout = branchLayoutReader.read(macheteFilePath);
```

Comment:
I'd call it 
```suggestion
        val currentBranchLayout = branchLayoutReader.read(macheteFilePath);
```
The word `managed` makes sense for _branch_, but not necessarily for the entire layout